offsite-activity-signup/src/Code.js

```diff
@@ -0,0 +1,458 @@
+  var startColumn = 1;
+  var numRows = rows.length;
+  var numColumns = rows[0].length; // NOTE: Assumes all rows equal length
+  sheet.getRange(startRow, startColumn, numRows, numColumns).setValues(rows);
+}
+
+/**
+ * Copies and randomizes an array
+ *
+ * @param {object[]} array - Array to shuffle
+ * @return {object[]} randomized copy of the array
+ */
+function shuffleArray_(array) {
+  var clone = array.slice(0);
+  for (var i = clone.length - 1; i > 0; i--) {
+    var j = Math.floor(Math.random() * (i + 1));
+    var temp = clone[i];
+    clone[i] = clone[j];
+    clone[j] = temp;
+  }
+  return clone;
+}
+
+/**
+ * Formats an number as an ordinal.
+ *
+ * See https://stackoverflow.com/questions/13627308/add-st-nd-rd-and-th-ordinal-suffix-to-a-number/13627586
+ *
+ * @param {number} i - Number to format
+ * @return {string} Formatted string
+ */
+function toOrdinal_(i) {
+  var j = i % 10;
+  var k = i % 100;
+  if (j == 1 && k != 11) {
+    return i + 'st';
+  }
+  if (j == 2 && k != 12) {
+    return i + 'nd';
+  }
+  if (j == 3 && k != 13) {
+    return i + 'rd';
+  }
+  return i + 'th';
+}
+
+/**
+ * Locates the sheet containing the form responses.
+ *
+ * @param {Spreadsheet} ss - Spreadsheet instance to search
+ * @return {Sheet} Sheet with form responses, undefined if not found.
+ */
+function findResponseSheetForForm_(ss) {
+  var formUrl = ss.getFormUrl();
+  if (!ss || !formUrl) {
+    return undefined;
+  }
+  var sheets = ss.getSheets();
+  for (var i in sheets) {
+    if (sheets[i].getFormUrl() === formUrl) {
+      return sheets[i];
+    }
+  }
+  return undefined;
+}
+
+/**
+ * Fills an array with a value ([].fill() not supported in Apps Script.)
+ *
+ * @param {object[]} arr - Array to fill
+ * @param {number} length - Number of items to fill.
+ * @param {object} value - Value to place at each index.
+ * @return {object[]} the array, for chaining purposes
+ */
+function fillArray_(arr, length, value) {
+  for (var i = 0; i < length; ++i) {
+    arr[i] = value;
+  }
+  return arr;
+}
+
+/**
+ * Creates an fills an array with numbers in the range [start, end].
+ *
+ * @param {number} start - First value in the range, inclusive
+ * @param {number} end - Last value in the range, inclusive
+ * @return {number[]} Array of values representing the range
+ */
+function range_(start, end) {
+  var arr = [start];
+  var i = start;
+  while (i < end) {
+    arr.push(i += 1);
+  }
+  return arr;
+}
+
+/**
+ * Transposes a matrix/2d array. For cases where the rows
+ * are not the same length, `fillValue` is used where no other
+ * value would otherwise be present.
+ *
+ * @param {object[][]} arr - 2D array to transpose
+ * @param {object} fillVaue - Placeholder for undefined values created as a result of the transpose.
+ *     Only required if rows aren't all of equal length.
+ * @return {object[][]} New transposed array
+ */
+function transpose_(arr, fillValue) {
+  var transposed = [];
+  arr.map(function(row, rowIndex) {
```

Comment:
Optional: I prefer `forEach` over `map` when the return value doesn't matter. Functionally the same, and maybe I'm being too pedantic. Alternatively just use a `reduce` instead.